Assets/Scripts/Framework/InputParser/PlayerInputParser.cs

```diff
@@ -0,0 +1,44 @@
+using System;
+using UnityEngine;
+using UnityEngine.InputSystem;
+
+namespace DerailedDeliveries.Framework.InputParser
+{
+    /// <summary>
+    /// A class responsible for parsing player input and invoking events based on the input.
+    /// </summary>
+    [RequireComponent(typeof(PlayerInput))]
+    public class PlayerInputParser : MonoBehaviour
+    {
+        /// <summary>
+        /// An event invoked when the player pressed the interact button.
+        /// </summary>
+        public Action OnInteract;
+
+        /// <summary>
+        /// An event invoked when the player gives input to move.
+        /// Note that this event is only called when the input changes and not every frame.
+        /// </summary>
+        public Action<Vector2> OnMove;
```

Comment:
Perhaps add a variable which stores the Vector2 as well so that if something subscribes to this delegate it can request the last value.